File: advent_of_java/src/main/java/Day1.java
import java.io.File;
import java.io.FileNotFoundException;
import java.util.ArrayList;
import java.util.Scanner;

public class Day1 {

    public static int solveDay1A() throws FileNotFoundException {
        File file = new File("Inputs/Day1.txt");
        Scanner scanner = new Scanner(file);

        int previousNum = Integer.parseInt(scanner.nextLine());
        int increases = 0;

        while(scanner.hasNext()){

            int currentNum = Integer.parseInt(scanner.nextLine());

            if( currentNum > previousNum){
                increases++;
            }
            previousNum = currentNum;
        }
        return increases;
    }

    public static int solveDay1B() throws FileNotFoundException {
        File file = new File("Inputs/Day1.txt");
        Scanner scanner = new Scanner(file);

        int increases = 0;
        ArrayList<Integer> numbers = new ArrayList<>();

        //init
        numbers.add(Integer.parseInt(scanner.nextLine()));
        numbers.add(Integer.parseInt(scanner.nextLine()));
        numbers.add(Integer.parseInt(scanner.nextLine()));

        int previousSum = numbers.get(0) + numbers.get(1) + numbers.get(2);

        while(scanner.hasNext()){

            numbers.remove(0);
            numbers.add(Integer.parseInt(scanner.nextLine()));

            int currentSum = numbers.get(0) + numbers.get(1) + numbers.get(2);

            if( currentSum > previousSum){
                increases++;
            }
            previousSum = currentSum;
        }
        return increases;
    }
}

File: advent_of_net/Day1.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace advent_of_net
{
    class Day1
    {


        public static int SolveDay1A()
        {
            string[] text = System.IO.File.ReadAllLines(@"..\..\..\..\Inputs\Day1.txt");

            int previousNum = Int32.Parse(text[0]);
            int increases = 0;

            for(int i = 1; i<text.Length; i++)
            {
                int currentNum = Int32.Parse(text[i]);

                if(currentNum > previousNum)
                {
                    increases++;
                }

                previousNum = currentNum;
            }
            return increases;
        }

        public static int SolveDay1B()
        {
            string[] text = System.IO.File.ReadAllLines(@"..\..\..\..\Inputs\Day1.txt");

            int previousSum = Int32.Parse(text[0]) + Int32.Parse(text[1]) + Int32.Parse(text[2]);
            int increases = 0;

            for (int i = 1; i < text.Length-2; i++)
            {
                int currentsum = Int32.Parse(text[i]) + Int32.Parse(text[i+1]) + Int32.Parse(text[i+2]);

                if (currentsum > previousSum)
                {
                    increases++;
                }

                previousSum = currentsum;
            }
            return increases;
        }
    }
}
